src/Voron/StorageEnvironmentOptions.cs

```diff
@@ -675,18 +676,19 @@ void Rename()
                     }
                 }
 
-                throw new InvalidOperationException("Unable to create scratch file " + name + ", even after trying multiple times." , err);
+                throw new InvalidOperationException("Unable to create temporary mapped file " + name + ", even after trying multiple times.", err);
+            }
+
+            public override AbstractPager CreateScratchPager(string name, long initialSize)
+            {
+                return GetMemoryMapPagerWithPossibleFileRename(name, initialSize, GetMemoryMapPager);
             }
 
             // This is used for special pagers that are used as temp buffers and don't 
             // require encryption: compression, recovery, lazyTxBuffer.
             public override AbstractPager CreateTemporaryBufferPager(string name, long initialSize)
             {
-                var scratchFile = TempPath.Combine(name);
-                if (File.Exists(scratchFile.FullPath))
-                    File.Delete(scratchFile.FullPath);
-
-                var pager = GetMemoryMapPagerInternal(this, initialSize, scratchFile, deleteOnClose: true);
+                var pager = GetMemoryMapPagerWithPossibleFileRename(name, initialSize, GetMemoryMapPagerInternal);
```

Comment:
Given that they are so similar, is there a reason not to just call CreateScratchPager here and avoid `GetMemoryMapPagerWithPossibleFileRename` ?

Reply:
going to implement this in this manner.

Reply:
https://github.com/ravendb/ravendb/pull/6648

Reply:
Wasn't aware of that, good point